### ISSUE TITLE
Mount emptyDir with medium: Memory to  control plan EE container

### ISSUE THREAD

**Issue by kdelee** [ansible/awx#14782]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

Enhancement to Existing Feature

### Feature Summary

As ansible output streams back from receptor, it is temporarily buffered to `/tmp` 

If this directory is not mounted to anything, AFAIK it behaves like a container would when otherwise running on a host -- it writes to the container's writable layer. Where that writable layer is depends on what storage driver is being used by the underlying container engine. Best docs I've found are from docker https://docs.docker.com/storage/storagedriver/#:~:text=When%20you%20create%20a%20new,this%20thin%20writable%20container%20layer. 

tl;dr writing to this layer can look like writing to the underlying worker node disk, and can be slow/put lots of pressure on the disk. 

On a VM, /tmp is often a tmpfs, which is memory backed. To get similar behavior in k8s, we could use a memoyr backed emptyDir. @fosterseth indicated the right container where receptor is buffering events to is the ee container in the task pod, where it stays until the dispatcher reads those events and enqueues them in Redis for the callback receiver to process.

https://github.com/ansible/awx-operator/blob/66f03c3c16431cb4f1483ae4d1d630b41721f145/roles/installer/templates/deployments/task.yaml.j2#L281-L336

Idea would be to add a volume mount to /tmp and indicate it is to be backed by Memory
```   
volumeMounts:
    - mountPath: /tmp
      name: receptor-tmp-volume
  volumes:
  - name: receptor-tmp-volume
    emptyDir:
      medium: Memory
```

Right thing to do is deploy it both ways, and run a workload that produces many events. Observe:
 - overall latency
 - event processing latency
 - overall node pressure (disk i/o, cpu, memory usage)

And determine if the memory backed empty dir is a net win.

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [X] Other

### Steps to reproduce

Run jobs that produce lots of output

### Current results

tbd, need to measure

### Sugested feature result

Reduce latency, disk i/o, cpu context switches

**Comment by Avilir** [comment]
Mounting a volume to the ee container (any kind of volume)  can improve performance, since container layer is COW (copy on write)  and it is slower then using memory or even regular disk filesystem.
Since we don't need to preserve the data that saved on /tmp, it will be better to use tmpfs (in memory) as volume type for better performance.

**Comment by kdelee** [comment]
Trying this out here https://github.com/ansible/awx-operator/compare/devel...kdelee:awx-operator:emptyDir-receptor-tmp?expand=1

Everything "just worked" on local deployment using `kind` except I did hit issue where I needed to raise 
```
sysctl -w fs.inotify.max_user_watches=2099999999
sysctl -w fs.inotify.max_user_instances=2099999999
sysctl -w fs.inotify.max_queued_events=2099999999
```
w/o which receptor didn't want to write files. (got error that specifically called out `failed to create fsnotify watcher: too many open files` But I have a feeling thats more do to do with my laptop/running too many things (kind cluster + docker dev env of gateway + awx + million chrome tabs).

Next TODO is test in more production like setting
